### PR TITLE
[Review request] Preparing for archive on Zenodo

### PR DESCRIPTION
**Pull request by Wu-Urbanek**
Please review the repository, especially the script. 
This branch is for the review purpose, the main branch is still the master.

### CODE REVIEW

File: cldf/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+pluggy==0.13.1
+purl==1.5
+py==1.8.1
+pybtex==0.22.2
+pycldf==1.14.1
+pyclts==2.0.0
+pyconcepticon==2.6.0
+pycountry==19.8.18
+pyglottolog==3.1.0
+-e git+https://github.com/lexibank/pylexibank.git@1e4a2f54afe7aee2e613f9c4bcb36f7d79f422fc#egg=pylexibank
```
Comment (chrzyki) [inline cldf/requirements.txt]: I'd go for the latest official release (2.6.0).

File: cldf/sources.bib [hunk 2]
```diff
@@ -0,0 +1,7 @@
+@misc{Bodt2019b,
+    author = {Aaley and Bodt, Uday Raj; and A., Timotheus},
```
Comment (chrzyki) [inline cldf/sources.bib]: The name order looks wrong I think?

File: cldf/requirements.txt (hunk 1 above)
Comment (Wu-Urbanek) [reply cldf/requirements.txt]: pylexibank?

Comment (chrzyki) [reply cldf/requirements.txt]: Yep!

**Comment by Wu-Urbanek** [comment]
Question, should I change cldfbench_aaleykusunda.py to lexibank_aaleykusunda.py ?

File: NOTES.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+This repository contains 230 Kusunda words according to the Concept list of the study "Laurent Sagart, Jacques, Guillaume, Yunfan Lai, and Johann-Mattis List (2019): Sino-Tibetan Database of Lexical Cognates. Jena, Max Planck Institute for the Science of Human History."
```
Comment (LinguList) [inline NOTES.md]: Come on, Mei-Shin, you can do this better. Just write:

```
This repository contains 230 words from the nearly extinct Kusunda language. These words were elicited with help of the [250 item list](https://concepticon.clld.org/contributions/Sagart-2019-250) underlying the study of [Sagart et al. (2019)](https://doi.org/10.1073/pnas.1817972116) on Sino-Tibetan languages.
```

File: cldf/sources.bib [hunk 4]
```diff
@@ -0,0 +1,7 @@
+@misc{Bodt2019b,
```
Comment (LinguList) [inline cldf/sources.bib]: Please quote the CALC blog as a journal: 

```
Cite this article as: Uday Raj Aaley and Timotheus A. Bodt, “New Kusunda data: A list of 250 concepts,” in Computer-Assisted Language Comparison in Practice, 08/04/2020, https://calc.hypotheses.org/2414.
```

This is an article, as the blog has an ISSN, so it is like a journal. See here: https://calc.hypotheses.org/
it appeared today.

File: cldfbench_aaleykusunda.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+class CustomLanguage(Language):
+    Location = attr.ib(default=None)
+    Remark = attr.ib(default=None)
+
+
+class Dataset(BaseDataset):
+    dir = pathlib.Path(__file__).parent
+    id = "aaleykusunda"
+    language_class = CustomLanguage
+    form_spec = FormSpec(
```
Comment (LinguList) [inline cldfbench_aaleykusunda.py]: Did you check and make sure that these brackets really occur in the source? Or the separators? If they don't occur, you don't need the formspec.

File: cldfbench_aaleykusunda.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        first_form_only=True
+    )
+
+    def cmd_makecldf(self, args):
+        # add bib
+        args.writer.add_sources()
+        args.log.info('added sources')
+
+        # add concept 
+        concepts=args.writer.add_concepts(id_factory=lambda c: c.id.split("-")[-1] + "_" + slug(c.english),
```
Comment (LinguList) [inline cldfbench_aaleykusunda.py]:
```
concepts = args.writer...
```

This is not inside a function, so please use spaces around the `=`

File: cldfbench_aaleykusunda.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                                        lookup_factory='Name')
+        # fix concept (concepticon source, data concept)
+        fix_concept = [('the barley (tibetan or highland)','the barley (Tibetan or highland)'),
+                       ('to plant (vegetals, rice)','to plant (vegetables, rice)')
+                    ]
+        for c in fix_concept:
+            if c[0] in concepts.keys():
```
Comment (LinguList) [inline cldfbench_aaleykusunda.py]:
```python
if c[0] in concepts:
```
The keys is not needed.

I would say: you do not even need to check, as we assume that you have exhaustively assembled all concepts to be fixed.

Why don't you write just two lines:

```
concepts["the barley (tibetan or highland)"] = concepts["the barley (Tibetan or highland)"]
concepts["to plant (vegetals, rice)"] = concepts["to plant (vegetables, rice)"]
```

`concepts` is a lookup, you don't have to delete anything from the dictionary, it is not important.

File: cldfbench_aaleykusunda.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+                concepts[c[1]]=concepts[c[0]]
+                del concepts[c[0]]
+        args.log.info('added concepts')
+
+        # add language
+        languages=args.writer.add_languages(lookup_factory='Name')
+        args.log.info('added languages')
+
+        # read in data
+        data=self.raw_dir.read_csv('Kusunda_2019_250_lexical_items.tsv',
```
Comment (LinguList) [inline cldfbench_aaleykusunda.py]: again, please write ` = `, outside of functions.

File: cldfbench_aaleykusunda.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+        # add language
+        languages=args.writer.add_languages(lookup_factory='Name')
+        args.log.info('added languages')
+
+        # read in data
+        data=self.raw_dir.read_csv('Kusunda_2019_250_lexical_items.tsv', 
+                                 delimiter='\t',dicts=True)
+        # add data
+        for gloss, entry in pb(list(enumerate(data)), desc='cldfify', total=len(data)):
```
Comment (LinguList) [inline cldfbench_aaleykusunda.py]: `enumerate` already makes this an iterable. The data is an iterable, so you don't need tow rite `list(`.

File: cldfbench_aaleykusunda.py [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+                del concepts[c[0]]
+        args.log.info('added concepts')
+
+        # add language
+        languages=args.writer.add_languages(lookup_factory='Name')
+        args.log.info('added languages')
+
+        # read in data
+        data=self.raw_dir.read_csv('Kusunda_2019_250_lexical_items.tsv', 
+                                 delimiter='\t',dicts=True)
```
Comment (LinguList) [inline cldfbench_aaleykusunda.py]: `, ` comma+space, this is normal style in Python, never leave no space after a comma, if something follows.

File: raw/sources.bib [hunk 11]
```diff
@@ -0,0 +1,7 @@
+@MISC{Bodt2019b,
```
Comment (LinguList) [inline raw/sources.bib]: see above, please change this to the blog post.

File: metadata.json [hunk 12]
```diff
@@ -0,0 +1,9 @@
+{
+    "id": "aaleykusunda",
+    "title": "A new wordlist of 250 items from the Kusunda language",
+    "description": null,
+    "license": "https://creativecommons.org/licenses/by-nc/4.0/",
+    "url": "https://zenodo.org/record/3377537#.XoXVZW7RZTY",
+    "citation": "Aaley, Uday Raj and Bodt, Timothaeus Adrianus (2019). A new wordlist of 250 items from the Kusunda language. London: School of Oriental and African Sciences.",
```
Comment (LinguList) [inline metadata.json]: Citation should be the blogpost as well, please modify this. And change date to 2020.

**Review by LinguList — COMMENTED** [review]
> Question, should I change cldfbench_aaleykusunda.py to lexibank_aaleykusunda.py ?

Yes, please.

File: raw/sources.bib (hunk 11 above)
Comment (Wu-Urbanek) [reply raw/sources.bib]: I use @article, is that ok? or @book?

File: cldf/sources.bib [hunk 13]
```diff
@@ -0,0 +1,9 @@
+@article{Bodt2019b,
+    author = {Aaley, Uday Raj and Bodt, Timotheus A.},
+    date = {2020},
+    day = {08},
```
Comment (LinguList) [inline cldf/sources.bib]: Please don't use date day, month in bibtex, they are barely recognized. Treat this as a normal journal, with volume (3), as this is the third year of the blog, and give it the number of the month, as we have monthly submissions: 

```
volume = {3},
number = {4},
year = {2020},
date = {2020-04-08}, # this is the correct way to display dates
```

File: lexibank_aaleykusunda.py [hunk 14]
```diff
@@ -0,0 +1,60 @@
+import re
```
Comment (LinguList) [inline lexibank_aaleykusunda.py]: why import re, csv, and requests?

File: lexibank_aaleykusunda.py [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+import csv
+import attr
+import pathlib
+import requests
+from clldutils.misc import slug
+from pylexibank import Dataset as BaseDataset
+from pylexibank.util import progressbar as pb
+from pylexibank import Language, CONCEPTICON_CONCEPTS
+from pylexibank.forms import FormSpec
+from collections import defaultdict
```
Comment (LinguList) [inline lexibank_aaleykusunda.py]: why import defaultdict?

File: lexibank_aaleykusunda.py [hunk 16]
```diff
@@ -0,0 +1,60 @@
+import re
+import csv
+import attr
+import pathlib
+import requests
+from clldutils.misc import slug
+from pylexibank import Dataset as BaseDataset
+from pylexibank.util import progressbar as pb
```
Comment (LinguList) [inline lexibank_aaleykusunda.py]: `from pylexibank import progressbar`

File: lexibank_aaleykusunda.py [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+from pylexibank import Language, CONCEPTICON_CONCEPTS
+from pylexibank.forms import FormSpec
```
Comment (LinguList) [inline lexibank_aaleykusunda.py]: `from pylexibank import FormSpec`

File: lexibank_aaleykusunda.py [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+
+        # add language
+        languages = args.writer.add_languages(lookup_factory='Name')
+        args.log.info('added languages')
+
+        # read in data
+        data = self.raw_dir.read_csv('Kusunda_2019_250_lexical_items.tsv',
+                                     delimiter='\t', dicts=True)
+        # add data
+        for gloss, entry in pb(enumerate(data),
```
Comment (LinguList) [inline lexibank_aaleykusunda.py]: wait, why enumerate now? You don't need to do that, just say `for row in data`

File: raw/sources.bib [hunk 19]
```diff
@@ -0,0 +1,9 @@
+@article{Bodt2019b,
```
Comment (LinguList) [inline raw/sources.bib:1]: wrong place, see my requests for representing this bibtex above, please.

**Review by LinguList — CHANGES_REQUESTED** [review]

File: raw/sources.bib [hunk 20]
```diff
@@ -1,9 +1,10 @@
...
-  month = {apr},
-  day = {08},
-  url	= {https://calc.hypotheses.org/2414},
+	title        = {New Kusunda data: A list of 250 concepts},
+	author       = {Aaley, Uday Raj and Bodt, Timotheus A.},
+	year         = {2020},
+	journal      = {Computer-Assisted Language Comparison in Practice},
+	volume       = {3},
+	url          = {https://calc.hypotheses.org/2414},
+	date         = {2020-04-08}
```
Comment (LinguList) [inline raw/sources.bib:9]: Number is missing:

```
number = {4},
```

**Review by LinguList — APPROVED** [review]
Please only add the year, afterwards, it can be merged. 
I hope you don't mind that I criticized the code in such detail. But it is important to pay attention to details, like libraries not imported, etc. Precision is important here...
But we're almost there.

File: raw/sources.bib (hunk 20 above)
Comment (Wu-Urbanek) [reply raw/sources.bib:9]: It's ok to criticize, although I was a bit confuse about the "wrong place". But all good now, I found the right file. So before I actually push the PR. The full bib is now: 
@article{Bodt2019b,
	title        = {New Kusunda data: A list of 250 concepts},
	author       = {Aaley, Uday Raj and Bodt, Timotheus A.},
	year         = {2020},
	journal      = {Computer-Assisted Language Comparison in Practice},
	volume       = {3},
        number   = {4},
	url          = {https://calc.hypotheses.org/2414},
	date         = {2020-04-08}. # keep it or remove
}

**Comment by LinguList** [comment]
Yes, this is exactly what I meant.

**Pull request merged**